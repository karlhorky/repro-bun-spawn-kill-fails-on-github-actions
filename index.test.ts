import { sleep, spawn } from 'bun';
import { expect, test } from 'bun:test';

test('Spawn and kill two child processes', async () => {
  const childProcess1 = spawn(['python3', '-m', 'http.server', '8000'], {
    stderr: 'pipe',
  });
  // const childProcess1 = spawn(['yarn', 'run', 'dev'], {
  //   stderr: 'pipe',
  // });
  console.log('childProcess1 started with PID', childProcess1.pid);

  const stdOutPromise = new Response(childProcess1.stdout).text();

  await sleep(2000);

  childProcess1.kill();
  const childProcess1ExitCode = await childProcess1.exited;
  console.log('childProcess1 exited with exit code', childProcess1ExitCode);

  expect(await stdOutPromise).toMatchSnapshot();

  const childProcess2 = spawn(['python3', '-m', 'http.server', '8000'], {
    stderr: 'pipe',
  });
  // const childProcess2 = spawn(['yarn', 'run', 'dev'], {
  //   stderr: 'pipe',
  // });
  console.log('childProcess2 started with PID', childProcess2.pid);
  childProcess2.kill();
  const childProcess2ExitCode = await childProcess2.exited;
  console.log('childProcess2 exited with exit code', childProcess2ExitCode);
});
