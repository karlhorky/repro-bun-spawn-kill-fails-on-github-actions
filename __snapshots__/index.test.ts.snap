// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`Spawn and kill two child processes 1`] = `
"yarn run v1.22.22
$ yarn tsx childProcess.ts
$ /Users/k/p/repro-bun-spawn-kill-fails-on-github-actions/node_modules/.bin/tsx childProcess.ts
Example app listening on port 3000
"
`;
